### DISCUSSION TITLE
WSL - Issues with cloning github repo using SSH

### DISCUSSION THREAD

**Discussion by gardgundersen44** [community/community#64582] [New to GitHub]
Hi, I am currently facing an issue when trying to clone a repo using SSH (instead of HTTPS). I have verified that HTTPS works, but I would like to use SSH if possible. I followed the Github links to setup SSH properly, but the problem occurs when I try to clone it in the Ubuntu terminal (using SSH). Here are the  links to the github guides (I chose the linux option):

(1) https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent
(2) https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

To make it easier for guys, this is what I have tried to do:
(1) Re-adding the SSH key to the agent, and synched the public key to Github
(2) Updated my driver for my Wi-Fi card to latest version
(3) Checked repository for any errors
(4) Verified the SSH connection using: ssh -T git@github.com (no errors)
(5) Verified that the key is added to the agent: ssh-add -l

This is what my console (in Ubuntu) outputs when I try to clone the repo:

```
"gard@DESKTOP-xx:~$ git clone git@github.com:some-project/CodePractice.git
Cloning into 'CodePractice'..." 
```

As you can see it just hangs after " Cloning into 'CodePractice'... ".  Also please take note that the SSH key is activated as you can see in one of the provided images. Please let me know if you need any additional information. 
<img width="410" alt="image1" src="https://github.com/community/community/assets/115529872/2b83d46c-2508-4b15-9a64-d42f0988bb31">
<img width="620" alt="image2" src="https://github.com/community/community/assets/115529872/66e6937a-ab05-41a0-9ec0-172549c0cc17">
<img width="683" alt="image3" src="https://github.com/community/community/assets/115529872/d9363ed4-ec3f-4f1d-b56c-0343affb2566">

Thank you for reading, have a nice day.
~ Gard

**Comment by RamtinKosari** [comment]
Hi daer @gardgundersen44 
Based on the information you provided, it seems like you have followed the necessary steps to set up SSH key authentication for your GitHub account. However, you are experiencing an issue when trying to clone a repository using SSH.

Here are a few additional troubleshooting steps you can try:

Check the repository URL: Make sure you are using the correct SSH URL for cloning the repository. The SSH URL should follow the format git@github.com:username/repository.git.

Verify your SSH configuration: Ensure that your SSH configuration is correctly set up. Open the ~/.ssh/config file (create it if it doesn't exist) and add the following lines:

```awk
Host github.com
  HostName github.com
  User git
  IdentityFile ~/.ssh/id_rsa
```
Replace id_rsa with the filename of your private key if it's different. Save the file and try cloning the repository again.

Disable firewall or antivirus: Temporarily disable any firewall or antivirus software on your system and try cloning the repository again. Sometimes these tools can interfere with SSH connections.

Check your network connection: Ensure that your internet connection is stable and not blocking SSH connections. If you are behind a proxy or firewall, make sure it allows SSH connections to GitHub.

Verify Git installation: Double-check that Git is installed correctly on your Ubuntu system. Run git --version to verify the installation and ensure you have the latest version.

If none of the above steps resolve the issue, it may be helpful to provide the specific error message or any additional information that is displayed after "Cloning into 'CodePractice'...". This will help in diagnosing the problem more accurately.

**Comment by LBF38** [comment]
Hi @gardgundersen44 
I don't know your full config. Have you got a VPN ? 
I know that I had some similar issues with SSH on WSL when cloning a repo. However, it throws an error in my case, didn't hang up long.

So, a solution would be to check your network configuration for the WSL instance.

Hope it helps ! Have a nice day !
